keadm/app/cmd/util/centosinstaller.go

```diff
@@ -140,41 +279,343 @@ func (c *CentOS) IsK8SComponentInstalled(component, defVersion string) (types.In
+	}
+
+	//Do an apt-get update
+	// 	# Set SELinux in permissive mode (effectively disabling it)
+	// setenforce 0
+	// sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config
+	cmd = &Command{Cmd: exec.Command("sh", "-c", "setenforce 0 && sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config")}
+	err := cmd.ExecuteCmdShowOutput()
+	fmt.Println("addK8SRepositoryAndUpdat4 ")
+	stdout := cmd.GetStdOutput()
+	errout := cmd.GetStdErr()
+	if err != nil || errout != "" {
+		return fmt.Errorf("%s", errout)
+	}
+	fmt.Println(stdout)
+	return nil
+}
+
 //InstallK8S will install kubeadm, kudectl and kubelet for the cloud node
 func (c *CentOS) InstallK8S() error {
-	fmt.Println("InstallK8S called")
+	//fmt.Println("InstallK8S called")
 	//Follow https://kubernetes.io/docs/setup/independent/install-kubeadm/
+	// yum install -y kubelet kubeadm kubectl --disableexcludes=kubernetes
+	// systemctl enable --now kubelet
+	k8sComponent := "kubeadm"
+	fmt.Println("Installing", k8sComponent, c.KubernetesVersion, "version")
+	if err := c.addK8SRepositoryAndUpdate(); err != nil {
+		return err
+	}
+
+	//Get the exact version string from OS repo, so that it can search and install.
+	chkKubeadmVer := fmt.Sprintf("yum list --showduplicates --disableexcludes=kubernetes %s| grep %s |awk '{print $2}' | cut -d'-' -f 1", k8sComponent, c.KubernetesVersion)
+	cmd := &Command{Cmd: exec.Command("sh", "-c", chkKubeadmVer)}
+	cmd.ExecuteCommand()
+	stdout := cmd.GetStdOutput()
+	errout := cmd.GetStdErr()
+	if errout != "" {
+		return fmt.Errorf("%s", errout)
+	}
+
+	fmt.Println("Expected K8S('", k8sComponent, "') version to install is", stdout)
+
+	//Install respective K8S components based on where it is being installed
+	k8sInst := fmt.Sprintf("yum install -y kubeadm-%s kubelet-%s kubectl-%s --disableexcludes=kubernetes", stdout, stdout, stdout)
+	cmd = &Command{Cmd: exec.Command("sh", "-c", k8sInst)}
+	err := cmd.ExecuteCmdShowOutput()
+	errout = cmd.GetStdErr()
+	if err != nil || errout != "" {
+		return fmt.Errorf("%s", errout)
+	}
+	fmt.Println(cmd.GetStdOutput())
+
+	enaKubelet := fmt.Sprintf("systemctl enable --now kubelet")
+	cmd = &Command{Cmd: exec.Command("sh", "-c", enaKubelet)}
+	err = cmd.ExecuteCmdShowOutput()
+	errout = cmd.GetStdErr()
+	if err != nil || errout != "" {
+		return fmt.Errorf("%s", errout)
+	}
+	fmt.Println(cmd.GetStdOutput())
+
+	fmt.Println(k8sComponent, "version", c.KubernetesVersion, "is installed in this Host")
+
 	return nil
 }
 
 //StartK8Scluster will do "kubeadm init" and cluster will be started
 func (c *CentOS) StartK8Scluster() error {
+	fmt.Println("StartK8sCluster")
+	var install bool
+	cmd := &Command{Cmd: exec.Command("sh", "-c", "kubeadm version")}
+	cmd.ExecuteCommand()
+	str := cmd.GetStdOutput()
+	if str != "" {
+		install = true
+	} else {
+		install = false
+	}
+	if install == true {
+		cmd := &Command{Cmd: exec.Command("sh", "-c", "swapoff -a && kubeadm init --pod-network-cidr 10.244.0.0/16")}
```

Comment:
Hi,I add --pod-network-cidr to kubeadm init because I would like to choose flannel as overlay networ. After  kubeedge init,the user should use "kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml"  ,then the node is ready. And in this yaml file the cidr is set to 10.244.0.0/16.
In the ubuntu installer ,I haven't found the network overlay configuration,without that ,the coredns pod  won't work and the node is notready.So what's your suggestion for the overlay network,should user choose there own network overlay after kubeedge init or should we add several overlay options to kubeedge init command? Thanks.

Reply:
@samy2019